packages/glimmer-syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -140,6 +140,10 @@ export default {
 
     delete path.depth;
 
+    if (original.match(/^this(\..+)?$/)) {
+      path.parts.unshift('this');
```

Comment:
The convention in Handlebars is to use an empty string internally for `this` so that you can differentiate it between `{{[this]}}`.